src/main/java/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentation.java

```diff
@@ -131,6 +130,17 @@ private boolean isDataLoaderCompatibleExecution(ExecutionContext executionContex
         return state.getApproach().beginExecutionStrategy(parameters, state.getState());
     }
 
+    @Override
+    public InstrumentationContext<ExecutionResult> beginDeferredField(InstrumentationDeferredFieldParameters parameters, InstrumentationState rawState) {
+        DataLoaderDispatcherInstrumentationState state = ofState(rawState);
+        //
+        // if there are no data loaders, there is nothing to do
+        //
+        if (state.hasNoDataLoaders()) {
+            return ExecutionStrategyInstrumentationContext.NOOP;
+        }
+        return state.getApproach().beginDeferredField(parameters, state.getState());
+    }
```

Comment:
lets make sure we have  test with a deferred field using a dataloader.

if the `dispatch()` never gets called - it will hang